WordPress/Classes/Models/BlogSettings+DateAndTimeFormat.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+
+/// In this extension, we implement several nested Enums (and helper setters / getters) aimed at simplifying
+/// the BlogSettings interface for handling writing date and time format properties.
+///
+extension BlogSettings {
+
+    /// Enumerates the Date format settings.
+    ///
+    enum DateFormat: String {
+        case MonthDY = "F j, Y"
+        case YMD = "Y-m-d"
+        case MDY = "m/d/Y"
+        case DMY = "d/m/Y"
+
+        /// Returns the sorted collection of all of the Localized Enum Titles.
+        ///
+        static var allTitles: [String] {
+            return allValues.flatMap { descriptionMap[$0] }
+        }
+
+        /// Returns the sorted collection of all of the possible Enum Values.
+        ///
+        static var allValues: [String] {
+            return [DateFormat.MonthDY.rawValue, DateFormat.YMD.rawValue,
+                    DateFormat.MDY.rawValue, DateFormat.DMY.rawValue]
+        }
+
+        /// Returns the localized description of the current enum value.
+        ///
+        var description: String {
+            return DateFormat.descriptionMap[rawValue]!
```

Comment:
`!` ⚠️ 😁 
A map might be more convenient, but using a switch is safer, since the compiler will warn if you add a new value in the future without providing a description.

Reply:
Actually, since you're never referencing the enum values directly, I wonder if it would be better to just have a dictionary of known formats

Reply:
I like the `Enum` approach since it also keeps things consistent with `BlogSettings+Discussion`, I can change it though if you feel strongly about it.